pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,41 @@
+package backuphelpers
+
+import "sync"
+
+type BackupVar interface {
+	GetBackupVars() map[string]string
```

Comment:
```suggestion
	GetBackupVars() BackupConfig
```

Reply:
I think we should keep the `map`, since the struct has other fields (e.g. sync.Mutex)

wdyt ?

Reply:
you can also reuse the embedded CRD itself, no need to invent a new one

Reply:
actually its a good idea, but i need to add new args anyways (ie. "data-dir", "backup-dir" )

Reply:
i used `global mutex` on the package, so that we move the struct around without the lock

Reply:
its not ideal i know, but its patch 

otherwise, i can make another struct for the config, and embed it into `BackupConfig` struct, 

the mutex will be on the outer struct only, but it used to mutate the inner struct state

Reply:
I don't understand why you need a global mutex, can you explain?

Reply:
when cloning the `BackupConfig`, will carry the mutex around 

I was using a `map` to get the args, but u asked for using the struct, thats why i wanted to use global mutex